libbeat/docs/shared-autodiscover.asciidoc

```diff
@@ -0,0 +1,80 @@
+[[configuration-autodiscover]]
+== Autodiscover
+
+Autodiscover allows you to watch for system changes and dynamically adapt settings to them, as they happen.
+This is specially useful when running your infrastructure on containers.
+
+When you run an application on containers it becomes a moving target to the monitoring system, Autodiscover
+allows you to automatically detect what's running and update settings to monitor it.
+
+You can define configuration templates for different containers, Autodiscover subsystem will use them
+to monitor services as they start running.
+
+[float]
+=== Configuration options
+
+Autodiscover settings are defined under {beatname_lc}.autodiscover field, to enable it you must configure
+a list of providers to be used.
+
+["source","yaml",subs="attributes"]
```

Comment:
TBH, I'm not sure this config example is useful without the details of the provider config. I'd suggest removing the entire "Configuration options" section because you don't describe the options. Then you can just tack on the above paragraph to the end of the intro:

````

...to monitor services as they start running.

You define autodiscover settings in the  +{beatname_lc}.autodiscover+ section of the +{beatname_lc}.yml+
config file. To enable autodiscover, you specify a list of providers.

+[float]
+=== Providers
`````

I'd like to make other changes to the intro, but that's probably good enough for now.